lib/text/ui_text_displayer.js

```diff
@@ -644,30 +716,73 @@ shaka.text.UITextDisplayer = class {
       }
     }
 
-    style.lineHeight = cue.lineHeight;
+    const computedPositionAlign = this.computeCuePositionAlignment_(cue);
+    const computedPosition = this.computeCuePosition_(cue);
+
+    // The positionAlign attribute is an alignment for the text container in
+    // the dimension of the writing direction.
+    if (computedPositionAlign === Cue.positionAlign.LEFT) {
+      style.alignItems = 'flex-start';
+    } else if (computedPositionAlign === Cue.positionAlign.RIGHT) {
+      style.alignItems = 'flex-end';
+    }
 
     // The position defines the indent of the text container in the
     // direction defined by the writing direction.
-    if (cue.position != null) {
-      if (cue.writingMode == Cue.writingMode.HORIZONTAL_TOP_TO_BOTTOM) {
-        style.paddingLeft = cue.position;
+    // TODO: factorize code since width/height, left/top and right/bottom
+    //       have same role, depending on the writing mode
+    if (cue.writingMode === Cue.writingMode.HORIZONTAL_TOP_TO_BOTTOM) {
+      // Horizontal writing
+      if (computedPositionAlign === Cue.positionAlign.LEFT) {
+        // Left side of cue box is positioned at 'computedPosition'
+        style.paddingLeft = computedPosition + '%';
+        style.width = (100 - computedPosition) + '%';
+      } else if (computedPositionAlign === Cue.positionAlign.RIGHT) {
+        // Right side of cue box is positioned at 'computedPosition'
+        style.paddingRight = (100 - computedPosition) + '%';
+        style.width = computedPosition + '%';
       } else {
-        style.paddingTop = cue.position;
+        // Position is 'center'
+        // Horizontal center of cue box is positioned at 'computedPosition'
+        if (computedPosition > 50) {
+          // Center of cue box is in the right part of the viewport
+          const width = (100 - computedPosition) * 2;
+          style.width = width + '%';
+          style.paddingLeft = (100 - width) + '%';
+        } else if (computedPosition < 50) {
+          // Center of cue box is in the left part of the viewport
+          const width = computedPosition * 2;
+          style.width = width + '%';
+          style.paddingRight = (100 - width) + '%';
+        }
```

Comment:
Shouldn't computedPosition==50 be handled in one of the above two cases?  Is it right to leave out styles in this case?

Reply:
In this particular case, the center of the box is centered at 50% of the viewport. So, there's really nothing to change to the style.
I might have missunderstood the specs, of course, but what I see in my tests seems valid.